include/alpaka/mem/buf/Traits.hpp

```diff
@@ -175,9 +177,10 @@ namespace alpaka
         TPlatform const& platform,
         TExtent const& extent = TExtent())
     {
-        if constexpr(hasMappedBufSupport<TPlatform>)
+        using PlatformForMappedAlloc = alpaka::Platform<TPlatform>;
```

Comment:
can you use `using Platform` like everywhere else?

Reply:
That was the initial version, until [this](https://github.com/alpaka-group/alpaka/pull/2120#discussion_r1312843301) was pointed out.

Reply:
My assumption was wrong. `Platform = Platform<T>` doesn't trigger `-Wshadow`. So feel free to follow @fwyzard's suggestion.

Reply:
`PlatformForMappedAlloc` should now be replaced by `Platform`.